src/Symfony/Bundle/FrameworkBundle/CacheWarmer/ValidatorCacheWarmer.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\CacheWarmer;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Cache\Adapter\ArrayAdapter;
+use Symfony\Component\Cache\Adapter\PhpArrayAdapter;
+use Symfony\Component\Cache\Adapter\ProxyAdapter;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Component\Validator\Mapping\Cache\Psr6Cache;
+use Symfony\Component\Validator\Mapping\Factory\LazyLoadingMetadataFactory;
+use Symfony\Component\Validator\Mapping\Loader\LoaderChain;
+use Symfony\Component\Validator\Mapping\Loader\LoaderInterface;
+use Symfony\Component\Validator\Mapping\Loader\XmlFileLoader;
+use Symfony\Component\Validator\Mapping\Loader\YamlFileLoader;
+use Symfony\Component\Validator\ValidatorBuilderInterface;
+
+/**
+ * Warms up XML and YAML validator metadata.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class ValidatorCacheWarmer implements CacheWarmerInterface
+{
+    private $validatorBuilder;
+    private $phpArrayFile;
+    private $fallbackPool;
+
+    /**
+     * @param ValidatorBuilderInterface $validatorBuilder
+     * @param string                    $phpArrayFile     The PHP file where metadata are cached.
+     * @param CacheItemPoolInterface    $fallbackPool     The pool where runtime-discovered metadata are cached.
+     */
+    public function __construct(ValidatorBuilderInterface $validatorBuilder, $phpArrayFile, CacheItemPoolInterface $fallbackPool)
```

Comment:
I only have a last suggestion: shouldn't we inject an instance of ``PhpArrayAdapter`` and add a method `getFallbackPool`` to it to avoid duplicating data in several places ?